qiskit_experiments/calibration/experiments/spectroscopy.py

```diff
@@ -0,0 +1,317 @@
+        - sigma: The standard deviation of the pulse, the default is 5 x duration.
+        - width: The width of the flat-top in the pulse, the default is 0, i.e. a Gaussian.
+
+        Args:
+            qubit: The qubit on which to run spectroscopy.
+            frequency_shifts: The frequencies to scan in the experiment.
+            unit: The unit in which the user specifies the frequencies. Can be one
+                of 'Hz', 'kHz', 'MHz', 'GHz'. Internally, all frequencies will be converted
+                to 'Hz'.
+
+        Raises:
+            ValueError: if there are less than three frequency shifts or if the unit is not known.
+
+        """
+        if len(frequency_shifts) < 3:
+            raise ValueError("Spectroscopy requires at least three frequencies.")
+
+        if unit not in self.__units__:
+            raise ValueError(f"Unsupported unit: {unit}.")
+
+        self._frequency_shifts = [freq * self.__units__[unit] for freq in frequency_shifts]
+
+        super().__init__([qubit], circuit_options=("amp", "duration", "sigma", "width"))
+
+    # pylint: disable=unused-argument
+    def circuits(self, backend: Optional["Backend"] = None, **circuit_options):
+        """
+        Create the circuit for the spectroscopy experiment. The circuits are based on a
+        GaussianSquare pulse and a frequency_shift instruction encapsulated in a gate.
+
+        Args:
+            backend: A backend object.
+            circuit_options: Key word arguments to run the circuits. The circuit options are
+                - amp: The amplitude of the GaussianSquare pulse, defaults to 0.1.
+                - duration: The duration of the GaussianSquare pulse, defaults to 10240.
+                - sigma: The standard deviation of the GaussianSquare pulse, defaults to one
+                    fith of the duration.
+                - width: The width of the flat top in the GaussianSquare pulse, defaults to 0.
+
+        Returns:
+            circuits: The circuits that will run the spectroscopy experiment.
+        """
+
+        amp = circuit_options.get("amp", 0.1)
+        duration = circuit_options.get("duration", 1024)
+        sigma = circuit_options.get("sigma", duration / 5)
+        width = circuit_options.get("width", 0)
+
+        drive = pulse.DriveChannel(self._physical_qubits[0])
+
+        circs = []
+
+        for freq_shift in self._frequency_shifts:
+            with pulse.build(name=f"Frequency shift{freq_shift}") as sched:
+                pulse.shift_frequency(freq_shift, drive)
+                pulse.play(pulse.GaussianSquare(duration, amp, sigma, width), drive)
+
+            gate = Gate(name="Spec", num_qubits=1, params=[])
+
+            circuit = QuantumCircuit(1)
+            circuit.append(gate, (0,))
+            circuit.add_calibration(gate, (self._physical_qubits[0],), sched)
+            circuit.measure_active()
+
+            circuit.metadata = {
```

Comment:
Can you add circuit options to metadata? I think knowing pulse setting would help to debug the experimental result.